### PR TITLE
RC v0.0.5

### PR DESCRIPTION
**Pull request by aburrell**
# Description

The v0.0.5 release improves standards, fixes bugs, and added a Dst forecasts to the supported data sets. This is an alpha release.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Ran unit tests locally against main and RC of pysat

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs (see CI block for link)
- [x] Upload the test distribution to the testPyPi server: https://test.pypi.org/project/pysatSpaceWeather/0.0.5/
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text: https://github.com/pysat/pysatSpaceWeather/wiki/v0.0.5
- [ ] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: .zenodo.json [hunk 1]
```diff
@@ -14,14 +27,19 @@
         "affiliation": "The University of Texas at Dallas",
```
Comment (aburrell) [inline .zenodo.json:27]: @rstoneback this needs to be updated.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 2]
```diff
@@ -323,116 +316,111 @@ def list_files(tag=None, inst_id=None, data_path=None, format_str=None):
...
+            pds_off = pds.DateOffset(days=1)
+            out_files.loc[out_files.index[-1] + pds_off] = out_files.values[-1]
+            out_files.loc[out_files.index[-1] + pds_off] = out_files.values[-1]
 
     return out_files
 
 
 def download(date_array, tag, inst_id, data_path, update_files=False):
-    """Routine to download F107 index data
+    """Fownload F107 index data from the appropriate repository.
```
Comment (aburrell) [inline pysatSpaceWeather/instruments/sw_f107.py:392]:
```suggestion
    """Download F107 index data from the appropriate repository.
```

File: pysatSpaceWeather/instruments/methods/ace.py [hunk 3]
```diff
@@ -179,6 +188,10 @@ def download(date_array, name='', tag='', inst_id='', data_path='', now=None):
     - File requested not available on server
 
     """
+    # Ensure a real data path has been provided
```
[the remark this answers is not in the document]
Comment (rstoneback) [reply pysatSpaceWeather/instruments/methods/ace.py:191]: I checked and the pysat template Instrument has the None default, which should be updated. The testing instruments list_files is ok.

File: pysatSpaceWeather/instruments/methods/ace.py [hunk 4]
```diff
@@ -179,6 +188,10 @@ def download(date_array, name='', tag='', inst_id='', data_path='', now=None):
...
+    if data_path is None:
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/methods/ace.py:192]: In practice this check should never be encountered as it was moved into pysat.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -11,12 +11,12 @@ jobs:
...
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.8", "3.9"]
         numpy_ver: [latest]
         include:
           - python-version: "3.7"
-            numpy_ver: "1.17"
+            numpy_ver: "1.18"
```
Comment (jklenzing) [inline .github/workflows/main.yml:19]:
```suggestion
            numpy_ver: "1.19"
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -11,12 +11,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.8", "3.9"]
```
Comment (jklenzing) [inline .github/workflows/main.yml:15]:
```suggestion
        python-version: ["3.9", "3.10"]
```

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -11,12 +11,12 @@ jobs:
...
         numpy_ver: [latest]
         include:
           - python-version: "3.7"
```
Comment (jklenzing) [inline .github/workflows/main.yml:18]:
```suggestion
          - python-version: "3.8"
```

**Comment by jklenzing** [comment]
@aburrell, I could put together a new logger pull to bring things into compliance with pysat/pysat#997 if needed

File: pysatSpaceWeather/instruments/methods/ace.py [hunk 8]
```diff
@@ -331,3 +346,75 @@ def load_csv_data(fnames, read_csv_kwargs=None):
...
+
+def ace_swepam_hourly_omni_norm(as_inst, speed_key='sw_bulk_speed',
+                                dens_key='sw_proton_dens',
+                                temp_key='sw_ion_temp'):
+    """Normalize ACE SWEPAM variables as described in the OMNI processing _[1].
+
+    Parameters
+    ----------
+    as_inst : pysat.Instrument
+        pysat Instrument object with ACE SWEPAM data
```
Comment (JonathonMSmith) [inline pysatSpaceWeather/instruments/methods/ace.py:359]:
```suggestion
        pysat Instrument object with ACE SWEPAM data
```
```suggestion
        pysat Instrument object with ACE SWEPAM data.
```

Comment (JonathonMSmith) [inline pysatSpaceWeather/instruments/methods/ace.py:359]:
```suggestion
        pysat Instrument object with ACE SWEPAM data
```
```suggestion
        pysat Instrument object with ACE SWEPAM data.
```

Comment (JonathonMSmith) [inline pysatSpaceWeather/instruments/methods/ace.py:359]:
```suggestion
        pysat Instrument object with ACE SWEPAM data.
```

Comment (JonathonMSmith) [inline pysatSpaceWeather/instruments/methods/ace.py:359]:
```suggestion
        pysat Instrument object with ACE SWEPAM data.
```

Comment (JonathonMSmith) [reply pysatSpaceWeather/instruments/methods/ace.py:359]: Not sure I completely grasp when periods are appropriate, but it looks like this function's parameter descriptions could use periods.

Comment (JonathonMSmith) [reply pysatSpaceWeather/instruments/methods/ace.py:359]: Not sure I completely grasp when periods are appropriate, but it looks like this function's parameter descriptions could use periods.

Comment (aburrell) [reply pysatSpaceWeather/instruments/methods/ace.py:359]: It's not necessary here, but I'll add the suggested changes.

**Comment by aburrell** [comment]
> Well, I only mention the periods because in the "ace_swepam_hourly_omni_norm" function there are none, but in ace_epam's load function descriptions like "Instrument tag (default='')." have periods.

That's a good reason, but since it's not a standard, I'm not going to do a sweep of the entire codebase for periods.

**Review by jklenzing — CHANGES_REQUESTED** [review]
minor update in tests, otherwise looking good.

**Comment by aburrell** [comment]
Pinging @rstoneback @jklenzing and @JonathonMSmith for approvals before Friday (new target release date).

File: pysatSpaceWeather/tests/test_methods_ace.py [hunk 9]
```diff
@@ -1,35 +1,106 @@
+#!/usr/bin/env python
+# Full license can be found in License.md
+# Full author list can be found in .zenodo.json file
+# DOI:10.5281/zenodo.3986138
+# ----------------------------------------------------------------------------
+"""Integration and unit test suite for ACE methods."""
+
+import pysat
```
Comment (jklenzing) [inline pysatSpaceWeather/tests/test_methods_ace.py:8]:
```suggestion
from packaging.version import Version
import pysat
```

File: pysatSpaceWeather/tests/test_methods_general.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+# ----------------------------------------------------------------------------
+"""Integration and unit test suite for ACE methods."""
+
+import numpy as np
+import pysat
+import pytest
+
+from pysatSpaceWeather.instruments.methods import general
+
+pysat_version_minor = int(pysat.__version__.split('.')[1])
```
Comment (jklenzing) [inline pysatSpaceWeather/tests/test_methods_general.py:14]:
```suggestion
```

File: pysatSpaceWeather/tests/test_methods_general.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+
+@pytest.mark.skipif(pysat_version_minor < 1,
```
Comment (jklenzing) [inline pysatSpaceWeather/tests/test_methods_general.py:17]:
```suggestion
@pytest.mark.skipif(pysat.__version__) < Version('3.0.2'),
```

**Review by jklenzing — CHANGES_REQUESTED** [review]
now with code suggestions!

**Review by jklenzing — APPROVED** [review]
Looks good! 😸

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 12]
```diff
@@ -294,21 +290,18 @@ def load(fnames, tag=None, inst_id=None):
     return data, meta
 
 
-def list_files(tag=None, inst_id=None, data_path=None, format_str=None):
-    """Return a Pandas Series of every file for F10.7 data
+def list_files(tag, inst_id, data_path, format_str=None):
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/sw_f107.py:293]: `tag`, `inst_id`, and `data_path` are keyword arguments.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 13]
```diff
@@ -487,8 +479,8 @@ def download(date_array, tag, inst_id, data_path, update_files=False):
                     # Create a local CSV file
                     data.to_csv(data_file, header=True)
     elif tag == 'prelim':
-        ftp = ftplib.FTP('ftp.swpc.noaa.gov')  # connect to host, default port
-        ftp.login()  # user anonymous, passwd anonymous@
+        ftp = ftplib.FTP('ftp.swpc.noaa.gov')  # Connect to host, default port
+        ftp.login()  # User anonymous, passwd anonymous
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/sw_f107.py:483]:
```suggestion
        # Connect to host, default port
        ftp = ftplib.FTP('ftp.swpc.noaa.gov')  
        
        # User anonymous, passwd anonymous
        ftp.login()  
```

File: pysatSpaceWeather/instruments/sw_kp.py [hunk 14]
```diff
@@ -55,13 +63,12 @@
 import numpy as np
 import os
 import pandas as pds
+import pysat
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/sw_kp.py:66]: pysat imports after more common, general packages

File: pysatSpaceWeather/instruments/sw_kp.py [hunk 15]
```diff
@@ -110,30 +113,26 @@ def init(self):
...
     return
 
 
 # ----------------------------------------------------------------------------
 # Instrument functions
 
 
-def load(fnames, tag=None, inst_id=None):
-    """Load Kp index files
+def load(fnames, tag, inst_id):
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/sw_kp.py:125]: keywords

File: pysatSpaceWeather/tests/test_methods_ace.py [hunk 16]
```diff
@@ -1,35 +1,105 @@
+#!/usr/bin/env python
+# Full license can be found in License.md
+# Full author list can be found in .zenodo.json file
+# DOI:10.5281/zenodo.3986138
+# ----------------------------------------------------------------------------
+"""Integration and unit test suite for ACE methods."""
+
+from packaging.version import Version
+import pysat
```
Comment (rstoneback) [inline pysatSpaceWeather/tests/test_methods_ace.py:9]: should pysat be imported lower?

File: pysatSpaceWeather/tests/test_methods_kp.py [hunk 17]
```diff
@@ -3,32 +3,42 @@
...
+
+        load_kwargs = {'date': self.test_time}
+        if Version(pysat.__version__) > Version('3.0.1'):
+            load_kwargs['use_header'] = True
+
+        self.testInst.load(**load_kwargs)
+
+        # Create Kp data
+        self.testInst.data.index = pds.DatetimeIndex(data=[
+            self.test_time + pds.DateOffset(hours=3 * i) for i in range(12)])
```
Comment (rstoneback) [inline pysatSpaceWeather/tests/test_methods_kp.py:38]: timedelta?

File: pysatSpaceWeather/tests/test_methods_ace.py (hunk 16 above)
Comment (aburrell) [reply pysatSpaceWeather/tests/test_methods_ace.py:9]: It's passing the flake8, so I think it's fine.

File: pysatSpaceWeather/instruments/sw_f107.py (hunk 13 above)
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_f107.py:483]: I don't think there's anything wrong with inline comments, do we advise against them in the style guide?

**Review by rstoneback — APPROVED** [review]
Looks good!

**Pull request merged**